src/adapters/curve/products/staking/curveStakingAdapter.ts

```diff
@@ -0,0 +1,313 @@
+import { SimplePoolAdapter } from '../../../../core/adapters/SimplePoolAdapter'
+import { ZERO_ADDRESS } from '../../../../core/constants/ZERO_ADDRESS'
+import {
+  IMetadataBuilder,
+  CacheToFile,
+} from '../../../../core/decorators/cacheToFile'
+import { NotImplementedError } from '../../../../core/errors/errors'
+import { filterMapSync } from '../../../../core/utils/filters'
+import { logger } from '../../../../core/utils/logger'
+import {
+  ProtocolDetails,
+  PositionType,
+  GetEventsInput,
+  MovementsByBlock,
+  GetAprInput,
+  GetApyInput,
+  GetTotalValueLockedInput,
+  TokenBalance,
+  ProtocolTokenApr,
+  ProtocolTokenApy,
+  ProtocolTokenTvl,
+  UnderlyingTokenRate,
+  Underlying,
+  ProtocolRewardPosition,
+  GetClaimableRewardsInput,
+  ProtocolAdapterParams,
+  TokenType,
+} from '../../../../types/adapter'
+import { Erc20Metadata } from '../../../../types/erc20Metadata'
+import { IProtocolAdapter } from '../../../../types/IProtocolAdapter'
+import { getPoolData } from '../../common/getPoolData'
+import { MetaRegistry__factory } from '../../contracts'
+import { CURVE_META_REGISTRY_CONTRACT } from '../pool/curvePoolAdapter'
+
+type CurveStakingAdapterMetadata = Record<
+  string,
+  {
+    protocolToken: Erc20Metadata
+    underlyingTokens: Erc20Metadata[]
+  }
+>
+const PRICE_PEGGED_TO_ONE = 1
+export class CurveStakingAdapter
+  extends SimplePoolAdapter
+  implements IMetadataBuilder
+{
+  productId = 'staking'
+
+  poolAdapter: IProtocolAdapter & IMetadataBuilder
+
+  constructor(params: ProtocolAdapterParams) {
+    super(params)
+
+    this.poolAdapter = params.adaptersController.fetchAdapter(
+      params.chainId,
+      params.protocolId,
+      'pool',
```

Comment:
Should be `this.productId`
```suggestion
      this.productId,
```